examples/pipeline_wavernn/main.py

```diff
@@ -0,0 +1,420 @@
+                # use mol mode
+                target = target.unsqueeze(-1)
+
+            loss = criterion(output, target)
+            sums["loss"] += loss.item()
+
+        avg_loss = sums["loss"] / len(data_loader)
+
+        metric = MetricLogger("validation")
+        metric("epoch", epoch)
+        metric("loss", avg_loss)
+        metric("time", time() - start)
+        metric.print()
+
+        return avg_loss
+
+
+def main(args):
+
+    devices = ["cuda" if torch.cuda.is_available() else "cpu"]
+
+    logging.info("Start time: {}".format(str(datetime.now())))
+
+    # Empty CUDA cache
+    torch.cuda.empty_cache()
+
+    melkwargs = {
+        "n_fft": args.n_fft,
+        "power": 1,
+        "hop_length": args.hop_length,
+        "win_length": args.win_length,
+    }
+
+    transforms = torch.nn.Sequential(
+        torchaudio.transforms.Spectrogram(**melkwargs),
+        # TODO Replace by torchaudio, once https://github.com/pytorch/audio/pull/593 is resolved
+        linear_to_mel(
+            sample_rate=args.sample_rate,
+            n_fft=args.n_fft,
+            n_mels=args.n_freq,
+            fmin=args.f_min,
+        ),
+        specgram_normalize(min_level_db=args.min_level_db),
+    )
+
+    train_dataset, val_dataset = gen_datasets_ljspeech(args, transforms)
+
+    loader_training_params = {
+        "num_workers": args.workers,
+        "pin_memory": False,
+        "shuffle": True,
+        "drop_last": False,
+    }
+    loader_validation_params = loader_training_params.copy()
+    loader_validation_params["shuffle"] = False
+
+    collate_fn = collate_factory(args)
+
+    train_loader = DataLoader(
+        train_dataset,
+        batch_size=args.batch_size,
+        collate_fn=collate_fn,
+        **loader_training_params,
+    )
+    val_loader = DataLoader(
+        val_dataset,
+        batch_size=args.batch_size,
+        collate_fn=collate_fn,
+        **loader_validation_params,
+    )
+
+    model = _WaveRNN(
+        upsample_scales=args.upsample_scales,
+        n_bits=args.n_bits,
+        sample_rate=args.sample_rate,
+        hop_length=args.hop_length,
+        n_res_block=args.n_res_block,
+        n_rnn=args.n_rnn,
+        n_fc=args.n_fc,
+        kernel_size=args.kernel_size,
+        n_freq=args.n_freq,
+        n_hidden=args.n_hidden,
+        n_output=args.n_output,
+        mode=args.mode,
+    )
+
+    if args.jit:
+        model = torch.jit.script(model)
+
+    model = torch.nn.DataParallel(model)
+    model = model.to(devices[0], non_blocking=True)
+
+    n = count_parameters(model)
+    logging.info(f"Number of parameters: {n}")
+
+    # Optimizer
+    optimizer_params = {
+        "lr": args.learning_rate,
+    }
+
+    optimizer = Adam(model.parameters(), **optimizer_params)
+
+    criterion = nn.CrossEntropyLoss() if args.mode == "waveform" else MoLLoss
+
+    best_loss = 10.0
+
+    load_checkpoint = args.checkpoint and os.path.isfile(args.checkpoint)
+
+    if load_checkpoint:
```

Comment:
since you are not using distributed setup, you can move the condition back in the if statement

Reply:
Do you mean:
`if args.checkpoint and os.path.isfile(args.checkpoint):`
instead of:
`load_checkpoint = args.checkpoint and os.path.isfile(args.checkpoint)`
`if load_checkpoint:`